jaq-fmts/src/write/raw0.rs

```diff
@@ -0,0 +1,37 @@
+use std::io::{Result, Error, ErrorKind, Write};
+use jaq_json::Val;
+
+struct NullTerminatedWriter<W>(W);
+
+impl<W: Write> NullTerminatedWriter<W> {
+    fn write_terminator(&mut self) -> Result<()> {
+        self.0.write_all(b"\0")
+    }
+}
+
+impl<W: Write> Write for NullTerminatedWriter<W> {
+    fn write(&mut self, buf: &[u8]) -> Result<usize> {
+        if buf.contains(&b'\0') {
+            return Err(Error::new(ErrorKind::InvalidData, "Cannot dump a string containing NUL with `--to raw0` or `--raw-output0`"));
+        }
+
+        self.0.write(buf)
+    }
+
+    fn flush(&mut self) -> Result<()> {
+        self.0.flush()
+    }
+}
+
+/// Write a value as CBOR.
+pub fn write(w: &mut dyn Write, pp: &jaq_json::write::Pp, v: &Val) -> Result<()> {
+    let mut w = NullTerminatedWriter(w);
+
+    if let Val::Str(b, _) = v {
+        w.write_all(b)?;
+    } else {
+        jaq_json::write::write(&mut w, pp, 0, v)?;
```

Comment:
Note that `jaq_json::write::write` never writes `'\0'`, because it escapes it to `"\\0"`.

Reply:
I didn't know! That helps a lot, and I'll gladly move the code to out of the raw0 module.